tests/runtests.py

```diff
@@ -426,6 +427,24 @@ def test_cmd_set_permission(self):
         res = gdb_start_silent_cmd("set-permission 0x1338000", target=target)
         self.assertNoException(res)
         self.assertIn("Unmapped address", res)
+
+        # Make sure set-permission command doesn't clobber any register
+        before = [
+            "starti",
+            "si",
+            "printf \"match_before\\n\"",
+            "info registers",
+            "printf \"match_before\\n\""
+        ]
+        after = [
+            "printf \"match_after\\n\"",
+            "info registers",
+            "printf \"match_after\\n\""
+        ]
+        res = gdb_run_cmd("set-permission $sp", before=before, after=after, target=target)
```

Comment:
why not `start_silent_cmd`?

Reply:
Resolved.

Reply:
@Grazfather I resolved it and unresolved it again. Please re-review my patch here:
https://github.com/hugsy/gef/pull/675/commits/63084b03357e29608285943bbd208f19ff5284ff